src/qt/optionsmodel.cpp

```diff
@@ -100,81 +85,132 @@ void OptionsModel::Init()
     if (!settings.contains("fShowMasternodesTab"))
         settings.setValue("fShowMasternodesTab", masternodeConfig.getCount());
 
+    // Main
+    setMainDefaultOptions(settings);
+
+// Wallet
+#ifdef ENABLE_WALLET
+    setWalletDefaultOptions(settings);
+#endif
+
+    // Network
+    setNetworkDefaultOptions(settings);
+    // Display
+    setDisplayDefaultOptions(settings);
+
+    language = settings.value("language").toString();
+}
+
+void OptionsModel::setMainDefaultOptions(QSettings& settings, bool reset){
     // These are shared with the core or have a command-line parameter
     // and we want command-line parameters to overwrite the GUI settings.
     //
     // If setting doesn't exist create it with defaults.
     //
     // If SoftSetArg() or SoftSetBoolArg() return false we were overridden
     // by command-line and show this in the UI.
-
     // Main
-    if (!settings.contains("nDatabaseCache"))
+    if (!settings.contains("nDatabaseCache") || reset)
         settings.setValue("nDatabaseCache", (qint64)nDefaultDbCache);
     if (!SoftSetArg("-dbcache", settings.value("nDatabaseCache").toString().toStdString()))
         addOverriddenOption("-dbcache");
 
-    if (!settings.contains("nThreadsScriptVerif"))
+    if (!settings.contains("nThreadsScriptVerif") || reset)
         settings.setValue("nThreadsScriptVerif", DEFAULT_SCRIPTCHECK_THREADS);
     if (!SoftSetArg("-par", settings.value("nThreadsScriptVerif").toString().toStdString()))
         addOverriddenOption("-par");
+}
 
-// Wallet
-#ifdef ENABLE_WALLET
-    if (!settings.contains("bSpendZeroConfChange"))
+void OptionsModel::setWalletDefaultOptions(QSettings& settings, bool reset){
+    if (!settings.contains("bSpendZeroConfChange") || reset)
         settings.setValue("bSpendZeroConfChange", false);
     if (!SoftSetBoolArg("-spendzeroconfchange", settings.value("bSpendZeroConfChange").toBool()))
         addOverriddenOption("-spendzeroconfchange");
-#endif
-    if (!settings.contains("nStakeSplitThreshold"))
+
+    if (!settings.contains("nStakeSplitThreshold") || reset)
         settings.setValue("nStakeSplitThreshold", 1);
+    if (reset){
```

Comment:
This isn't quite the behavior I would have expected.  I would have expected "reset" to take it back to the way it was before I started changing stuff and decided not to save.  e.g. if I change "Stake split threshold" to something; I can't get it to read back what I actually have saved without exiting and starting the wallet again.  I would have thought "reset" (specifically avoiding 'to default', as that's where the confusion lies) would have re-loaded what my saved settings were.

Now, it does say "to default", however what the default values is isn't very clear, and it goes ahead and saves it.  It should at least populate the values but require the user to 'save' them if they want to keep those defaults.

Also, I thought default stakesplitthreshold was 2000

Reply:
Ok good, have an user friendly solution for that.
The current behavior is quite straightforward, "Reset to default" returns the options of the screen back to the app's default values and "SAVE" stores the values that are on the specific screen.

What will do is add a "Clean unsaved changes" button which cleans the values that were not saved to the stored values. An invisible button that only appears when there are unsaved changes on the screen, the "SAVE" button ofc will store them and make the button invisible again.

And about stakesplitthreshold, that is actually a pre-existent bug, check optionsmodel.cpp, init method on a previous commit. It's set to 1 by default.. what did there was mainly a refactor of that crappy large monolithic method to be able to reset each screen to default separately without code duplication.
Will move the value to a constant and fix it here once for all, good catch 👍 .